src/clj/witan/send/send.clj

```diff
@@ -1,659 +1,14 @@
-                 (concat [(map name columns)])
-                 (csv/write-csv writer))))
-        (with-open [writer (io/writer (io/file (str dir "/Output_AY_Group.csv")))]
-          (let [columns [:calendar-year :ay-group :mean :std-dev :iqr :min :low-ci :q1 :median :q3 :high-ci :max]]
-            (->> (mapcat (fn [output year]
-                           (map (fn [[ay-group stats]]
-                                  (-> (medley/map-vals round stats)
-                                      (assoc :ay-group ay-group :calendar-year year)))
-                                (:total-in-send-by-ay-group output))) send-output (range initial-projection-year 3000))
-                 (map (apply juxt columns))
-                 (concat [(map name columns)])
-                 (csv/write-csv writer))))
-        (when run-charts
-          (with-open [writer (io/writer (io/file (str dir "/historic-data.csv")))]
-            (let [columns [:calendar-year :setting-1 :need-1 :academic-year-1 :setting-2 :need-2]
-                  headers (mapv name columns)
-                  rows (mapv #(mapv % columns) transitions-data)]
-              (csv/write-csv writer (into [headers] rows))))
-          (with-open [writer (io/writer (io/file (str dir "/valid-settings.csv")))]
-            (csv/write-csv writer valid-settings))
-          (with-open [writer (io/writer (io/file (str dir "/joiner-rates.csv")))]
-            (let [columns (into [] (keys joiner-ribbon-data))
-                  headers (mapv name columns)
-                  rows (ribbon-data-rows joiner-ribbon-data)]
-              (csv/write-csv writer (into [headers] rows))))
-          (with-open [writer (io/writer (io/file (str dir "/leaver-rates.csv")))]
-            (let [columns (into [] (keys leaver-ribbon-data))
-                  headers (mapv name columns)
-                  rows (ribbon-data-rows leaver-ribbon-data)]
-              (csv/write-csv writer (into [headers] rows))))
-          (with-open [writer (io/writer (io/file (str dir "/mover-rates.csv")))]
-            (let [columns (into [] (keys mover-ribbon-data))
-                  headers (mapv name columns)
-                  rows (ribbon-data-rows mover-ribbon-data)]
-              (csv/write-csv writer (into [headers] rows))))
-          (println "Producing charts...")
-          (with-open [in (io/input-stream (io/resource "send-charts.R"))]
-            (io/copy in (io/file "/tmp/send-charts.R")))
-          (sh/sh "Rscript" "--vanilla" "/tmp/send-charts.R" dir)
-          (run! #(io/delete-file (str dir "/" %) :quiet)
-                ["historic-data.csv" "valid-settings.csv" "joiner-rates.csv"
-                 "leaver-rates.csv" "mover-rates.csv"]))))
-    (report/write-send-report (str dir "/SEND_Log.md"))))
+   (-> (i/build-input-datasets (:project-dir config) (:file-inputs config) (:schema-inputs config))
+       (i/prepare-send-inputs (:transition-parameters config))
+       (r/run-send-model (:run-parameters config)))))
```

Comment:
Should this fn just be in main or something, just to dump send.clj? (I understand it may actually be here to avoid circular deps...?)